src/biogeophys/OzoneMod.F90

```diff
@@ -424,9 +445,120 @@ subroutine CalcOzoneUptake(this, bounds, num_exposedvegp, filter_exposedvegp, &
     end associate
 
   end subroutine CalcOzoneUptake
+    !-----------------------------------------------------------------------
+subroutine CalcOzoneUptakeLi2024OnePoint( &
+       forc_ozone, forc_pbot, forc_th, &
+       rs, rb, ram, &
+       tlai, tlai_old, pft_type, &
+       o3uptake, sabv)
+    !
+    ! !DESCRIPTION:
+    ! Calculates ozone uptake for a single point, for just sunlit or shaded leaves
+    !
+    ! !USES:
+    use shr_const_mod        , only : SHR_CONST_RGAS
+    use clm_time_manager     , only : get_step_size
+    !
+    ! !ARGUMENTS:
+    real(r8) , intent(in)    :: forc_ozone ! ozone partial pressure (mol/mol)
+    real(r8) , intent(in)    :: forc_pbot  ! atmospheric pressure (Pa)
+    real(r8) , intent(in)    :: forc_th    ! atmospheric potential temperature (K)
+    real(r8) , intent(in)    :: rs         ! leaf stomatal resistance (s/m)
+    real(r8) , intent(in)    :: rb         ! boundary layer resistance (s/m)
+    real(r8) , intent(in)    :: ram        ! aerodynamical resistance (s/m)
+    real(r8) , intent(in)    :: tlai       ! one-sided leaf area index, no burying by snow
+    real(r8) , intent(in)    :: tlai_old   ! tlai from last time step
+    integer  , intent(in)    :: pft_type   ! vegetation type, for indexing into pftvarcon arrays
+    real(r8) , intent(inout) :: o3uptake   ! ozone entering the leaf
+    real(r8) , intent(in)    :: sabv       ! patch solar radiation absorbed by vegetation (W/m**2)
+
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: dtime          ! land model time step (sec)
+    real(r8) :: dtimeh         ! time step in hours
+    real(r8) :: o3concnmolm3   ! o3 concentration (nmol/m^3)
+    real(r8) :: o3flux         ! instantaneous o3 flux (nmol m^-2 s^-1)
+    real(r8) :: o3fluxcrit     ! instantaneous o3 flux beyond threshold (nmol m^-2 s^-1)
+    real(r8) :: o3fluxperdt    ! o3 flux per timestep (mmol m^-2)
+    real(r8) :: heal           ! o3uptake healing rate based on % of new leaves growing (mmol m^-2)
+    real(r8) :: leafturn       ! leaf turnover time / mortality rate (per hour)
+    real(r8) :: decay          ! o3uptake decay rate based on leaf lifetime (mmol m^-2)
+    real(r8) :: o3_flux_threshold   ! threshold below which o3flux is set to 0 (nmol m^-2 s^-1)
+    real(r8) :: lai_thresh     !LAI threshold for growing season
+    ! o3:h2o resistance ratio from Li et al. 2024
+    real(r8), parameter :: ko3 = 1.51_r8
+
+    character(len=*), parameter :: subname = 'CalcOzoneUptakeOnePoint'
+    !-----------------------------------------------------------------------
+
+    ! convert o3 from mol/mol to nmol m^-3
+    o3concnmolm3 = forc_ozone * 1.e9_r8 * (forc_pbot/(forc_th*SHR_CONST_RGAS*0.001_r8))
```

Comment:
When you finalize this conversion (see my other comment), then replace any remaining hardwiring with parameter names. If the conversion parameters already exist elsewhere, please reuse them, and otherwise add them locally.

Reply:
add as local variables